src/net/tcp/stream.rs

```diff
@@ -70,12 +70,17 @@ impl TcpStream {
         Self { inner }
     }
 
-    /// Read some data from the stream into the buffer, returning the original buffer and
-    /// quantity of data read.
+    /// Read some data from the stream into the buffer.
+    ///
+    /// Takes ownership of the buffer for the duration of the Future.
+    ///
+    /// Returns the original buffer and quantity of data read.
     pub async fn read<T: BoundedBufMut>(&self, buf: T) -> crate::BufResult<usize, T> {
         self.inner.read(buf).await
     }
 
+    /// Read some data from the stream into a fixed, pre-registered buffer.
```

Comment:
I wouldn't say `fixed` in the first line where a simple/short description is intended. Where the name `fixed` comes from is a long story with its roots in the uring kernel driver itself and I think we try not to overuse that term. Better if we just say `Read some data from the stream into a registered buffer`.